modules/utxo-lib/src/inscriptions.ts

```diff
@@ -0,0 +1,66 @@
+import * as assert from 'assert';
+import { Payment, script as bscript } from 'bitcoinjs-lib';
+import { p2tr } from './payments';
+import { ecc as eccLib } from './';
+
+const OPS = bscript.OPS;
+const MAX_LENGTH_TAP_DATA_PUSH = 520;
+
+/**
+ * The max size of an individual OP_PUSH in a Taproot script is 520 bytes. This
+ * function splits inscriptionData into an array buffer of 520 bytes length.
+ * https://docs.ordinals.com/inscriptions.html
+ * @param inscriptionData
+ */
+function sliceInscriptionData(inscriptionData: Uint8Array) {
+  if (inscriptionData.length <= MAX_LENGTH_TAP_DATA_PUSH) {
+    return [Buffer.from(inscriptionData)];
+  }
+
+  const pushDataBuffers: Buffer[] = [];
+  for (let i = 0; i < inscriptionData.length; i += MAX_LENGTH_TAP_DATA_PUSH) {
+    const fixedLengthChunk = inscriptionData.slice(i, i + MAX_LENGTH_TAP_DATA_PUSH);
+    pushDataBuffers.push(Buffer.from(fixedLengthChunk));
+  }
+
+  return pushDataBuffers;
+}
+
+/**
+ * Generate an inscription address.
+ * @param userPubkey
+ * @param inscriptionData
+ * @param contentType
+ * @returns inscription address
+ */
+export function makeInscriptionAddress(
+  userPubkey: Uint8Array,
+  inscriptionData: Uint8Array,
+  contentType: string
+): string {
+  const dataPushBuffers = sliceInscriptionData(inscriptionData);
+  const pubKeyBuffer = Buffer.from(userPubkey);
+
+  const uncompiledScript = [
+    pubKeyBuffer,
+    OPS.OP_CHECKSIG,
+    OPS.OP_FALSE,
+    OPS.OP_IF,
+    Buffer.from('ord', 'ascii'),
+    OPS.OP_1,
+    Buffer.from(contentType, 'ascii'),
+    OPS.OP_0,
+    ...dataPushBuffers,
+    OPS.ENDIF,
+  ];
+
+  const compiledScript = bscript.compile(uncompiledScript);
+  const redeem: Payment = {
+    output: compiledScript,
+    depth: 0,
+  };
+  const payment = p2tr({ redeems: [redeem], pubkey: pubKeyBuffer }, { eccLib });
+
+  assert(payment.address, 'Failed to create inscription address');
+  return payment?.address;
```

Comment:
I'm a bit confused by this - the `assert` assumes that `payment` exists, but the `return` statement 
```suggestion
  assert(payment.output, 'Failed to create inscription output script');
  return payment.output;
```

Reply:
good catch, i'm asserting the wrong field here